cmake/modules/BuildOpentelemetry.cmake

```diff
@@ -13,7 +13,7 @@ function(build_opentelemetry)
                                -DWITH_JAEGER=ON
                                -DBUILD_TESTING=OFF
                                -DWITH_EXAMPLES=OFF
-                               -DCMAKE_BUILD_TYPE=Release)
+                               -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
```

Comment:
@zenomri I used Release as `cmake_build_type` so as we keep submodule verbosity and debug builds as I do not expect opentelemetry to require also be requiring to debug builds if Ceph is being so.
Is there some other consideration that you have?

Reply:
@ideepika  IMO, we should propagate the cmake_build_type, one more advantage of it is that if we would like the opentelemetry lib to expose debug symbols (I used it in profiling)